src/main/java/hudson/remoting/Launcher.java

```diff
@@ -633,7 +653,7 @@ public List<String> parseJnlpArguments() throws ParserConfigurationException, SA
         while (true) {
             URLConnection con = null;
             try {
-                con = JnlpAgentEndpointResolver.openURLConnection(agentJnlpURL, agentJnlpCredentials, proxyCredentials, sslSocketFactory, noCertificateCheck, null);
+                con = JnlpAgentEndpointResolver.openURLConnection(agentJnlpURL, null, agentJnlpCredentials, proxyCredentials, sslSocketFactory, hostnameVerifier);
```

Comment:
As elsewhere, putting mandatory user-specified arguments before optional ones, and putting user-specified arguments before computed things like SSL socket factories and hostname verifiers.